CONTRIBUTING.md

```diff
@@ -0,0 +1,85 @@
+# Contributing
+
+:warning: This project is very alpha, expect bugs!
+These instructions are intended for contributors to get a productive workflow when developing the plugin.
+
+This project follows [scalameta's contribution guidelines].
+Please read them for information about how to create good bug reports and submit pull requests.
+
+## Features
+
+- [x] Lint with Scalafix on compile
+- [x] Formatting with Scalafmt
+- [x] Go to definition in project sources using Semanticdb
+- [ ] Go to definition in dependencies
+- [ ] Red squigglies as you type
+- [ ] Auto completions
+- [ ] Signature helper (show parameter list)
+- [ ] Show type at position
+- [ ] Find references
+- [ ] Show docstring
+
+
+## Project structure
+- `language-server` contains a Scala implementation of a language server
+- `vscode-extension` contains a Visual Studio Code extension, implementing a client for the language server
+- `test-workspace` directory for manually testing the plugin locally
+
+
+## Running locally
+
+First you need to have the following applications installed
+
+- `git`
+- `sbt`
+- Visual Studio Code (`code` from the console)
+- `npm`
+
+To try out the language-server locally, it's best to keep two open terminal
+windows.
+
+```
+git clone https://github.com/scalameta/language-server.git
+cd language-server
+
+########
+# Step 1
+########
+sbt
+> language-server/publishLocal # publish your latest changes locally
+                               # keep this sbt shell session open, and
+                               # re-run publishLocal every time you
+                               # edit *.scala sources.
+
+########
+# Step 2
+########
+# Inside a new terminal window
+cd vscode-extension
+npm install
```

Comment:
I think that there should be also `npm run download-coursier` or `vscode:prepublish`, because coursier jar is assumed to be in the plugin's directory. 

I don't know about other npm scripts, what's `postinstall` for?

Reply:
For contributing, it's also useful to know how to run tests and what to expect from it. Sorry if it's something obvious in the npm/js/ts context, I'm just totally new to it 😅

Reply:
We currently don't have any test (currently trying to get them running in #12).

`postinstall` allows the vscode to setup itself after it has been installed. It's an automatic step that runs after `install` so it's not essential in the contribution guide I think.

Reply:
OK, I see. So no other npm tasks are of interest for a contributor?

Reply:
I would say that `npm install` and `npm run build` are enough for most cases.

Reply:
@olafurpg could you still add `npm run download-coursier` before `npm install`? Because I think without it coursier will be missing.

Reply:
Done @laughedelic I didn't need to run that command, but it's better to include more than too little 😄

Reply:
I think that coursier should be downloaded by the client on the first launch, so this pre-installation step won't be needed.

Reply:
I just added the coursier binary to the repo (it's only 20kb) so this step should no longer be needed. The binary was excluded for some reason

Reply:
I just noticed that you added coursier to the repo